python/ray/data/_internal/execution/operators/task_pool_map_operator.py

```diff
@@ -110,8 +110,35 @@ def _add_bundled_input(self, bundle: RefBundle):
     def progress_str(self) -> str:
         return ""
 
-    def base_resource_usage(self) -> ExecutionResources:
-        return ExecutionResources()
+    def min_max_resource_usage_bounds(
+        self,
+    ) -> Tuple[ExecutionResources, ExecutionResources]:
+        return self._min_resource_usage(), self._max_resource_usage()
+
+    def _min_resource_usage(self) -> ExecutionResources:
+        # Make sure the reserved resources are at least to allow one task.
+        return self.incremental_resource_usage()
+
+    def _max_resource_usage(self) -> ExecutionResources:
+        if self._inputs_complete:
+            # If the operator has already received all input data, we know it won't
+            # launch more tasks. So, we only need to reserve resources for the tasks
+            # that are currently running.
+            num_cpus_per_task = self._ray_remote_args.get("num_cpus", 0)
+            num_gpus_per_task = self._ray_remote_args.get("num_gpus", 0)
+            object_store_memory_per_task = (
+                self._metrics.obj_store_mem_max_pending_output_per_task or 0
+            )
+            resources = ExecutionResources.for_limits(
+                cpu=num_cpus_per_task * self.num_active_tasks(),
+                gpu=num_gpus_per_task * self.num_active_tasks(),
+                object_store_memory=object_store_memory_per_task
+                * self.num_active_tasks(),
+            )
+        else:
+            resources = ExecutionResources.for_limits()
+
+        return resources
```

Comment:
Let's generalize this to be base method in PhysicalOperator

Reply:
I don't think this is a sensible default for `PhysicalOperator`. The implementation makes sense for `TaskPoolOperator`, but I don't think we can assume it makes sense for all `PhysicalOperator` subclasses (e.g., the current all-to-all operator implementation)